### PR TITLE
Bug 1219629 - Implement *test_browser_clear_browsing_history.py* as an integration test in JavaScript

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=98b3610efb0ebff395277d322465fb1f884645f7

File: apps/system/test/marionette/lib/system.js [hunk 1]
```diff
@@ -389,6 +389,17 @@ System.prototype = {
     return frame;
   },
 
+  closeBrowser: function(url) {
```
Comment (albertopq) [inline apps/system/test/marionette/lib/system.js]: What is the url used for? I think this method should be named 'closeActiveApp' (or closeAppByUrl, if we use the parameter)

File: apps/system/test/marionette/browser_clear_browsing_history_test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    rocketbar = new Rocketbar(client);
+    settingsApp = new Settings(client);
+    actions = client.loader.getActions();
+    system.waitForFullyLoaded();
+  });
+
+  test('Clear browsing history', function() {
+    // Open then close web page.
+    var url = server.url('sample.html');
+    rocketbar.homescreenFocus();
```
Comment (albertopq) [inline apps/system/test/marionette/browser_clear_browsing_history_test.js]: You can use rocketbar.goToURL(url), which does exactly the same than these 3 lines.

File: apps/system/test/marionette/browser_clear_browsing_history_test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    system.closeBrowser(url);
+
+    // Open browser app.
+    client.apps.launch(search.URL);
+    client.apps.switchToApp(search.URL);
+
+    client.waitFor(function() {
+      return search.getHistoryResults().length === 1;
+    });
+    assert.equal(search.getHistoryResults().length, 1);
```
Comment (albertopq) [inline apps/system/test/marionette/browser_clear_browsing_history_test.js]: What's the value of this assertion after the previous waitFor?

File: apps/system/test/marionette/lib/system.js (hunk 1 above)
Comment (gmarty) [inline apps/system/test/marionette/lib/system.js]: I'd like to retain the word `browser` when we interact with web pages (See `system.gotoBrowser(url)`). I'll call it `closeBrowserByUrl` if that OK with you.

File: apps/system/test/marionette/browser_clear_browsing_history_test.js (hunk 3 above)
Comment (gmarty) [inline apps/system/test/marionette/browser_clear_browsing_history_test.js]: Oops, forgot to remove this line.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e25d3a44d8779e416d408c637abcc24a4743a9ca

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e25d3a44d8779e416d408c637abcc24a4743a9ca

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=47bfce208f78b7133ffbf2b9136ed0a45483d1aa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8b027cc9fd47c3684c9d00d582d67e873f0fa46d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7c478d832fadbc1cc7ef358bb9457fb4e5b8e97e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5d47cd46408cb718a816982ce66bc96a5d81140b

**Pull request merged**